autogpt/plugins/__init__.py

```diff
@@ -259,16 +259,8 @@ def scan_plugins(config: Config, debug: bool = False) -> List[AutoGPTPluginTempl
                     if key.startswith("__"):
                         continue
                     a_module = getattr(zipped_module, key)
-                    a_keys = dir(a_module)
 
-                    is_plugin = False
-                    if "_abc_impl" in a_keys:
-                        for base in a_module.__bases__:
-                            if base.__name__ == "AutoGPTPluginTemplate":
-                                is_plugin = True
-                                break
-
-                    if is_plugin and a_module.__name__ != "AutoGPTPluginTemplate":
+                    if issubclass(a_module, AutoGPTPluginTemplate) and a_module.__name__ != "AutoGPTPluginTemplate":
```

Comment:
my python might mislead me here, but doesn't this mean the class needs to exist as such? the class is currently a dependency of the plugins not a part of autogpt itself, (currently its listed in the dev-dependencies in requirements.txt, this change makes it a runtime dependency if i'm not mistaken)